### PR TITLE
[READY]Adds a new race: Ethereal (also adds wrappers for nutrition adjustment)

### PR DESCRIPTION
**Pull request by ghost**
:cl: Qustinnus / Floyd / Ethereal sprites by Space, is that it? / Alerts and food sprites by MrDroppodBringer
add: Adds Ethereal; a race which lives off of electricity and shines bright. If they are healthy they shine a bright green light, and the more damaged they are, the less they shine and the greyer they become. Their punches do burn damage and they are weak to blunt attacks!
They dont need to eat normal food and dont have nutrition, Instead they gain charge by going into borg rechargers, eating ethereal food or doign specific interactions.
refactor: all nutrition changes now go through a proc so we can override behavior
/:cl:

Adds Ethereal. A race which is essentialy inspired by Zoltan but then with different mechanics to make it fit with SS13 more. I'm trying to stay away from making them actually provide power but the change from nutrition to electricity seemed kind of fun.

They have the following specifications:

- They are walking lights; and shine bright and green if healthy, and get greyer and darker the more damaged they are. When dead they are just a grey unlit corpse.
- They do burn punch damage instead of blunt.
- They are weaker to blunt; especially when low on charge
- They have charge instead of nutrition. Charge is gained by some interactions or by going into a borg charger. or eating new ethereal food. 
- They are slightly stronger to shock damage, and gain charge from it a bit; however, it is still deadly to them and thus they aren't free insulated gloves.

[VIDEO](https://streamable.com/h0cuo)

Video of man eating ethereal food https://streamable.com/66o8g

ethereal charging up his charge https://streamable.com/dewkm

Feel free to give feedback or cool ideas that you think might benefit this new race!

TODO:

- [x] Fix jumpsuits being invisible on ethereal for some fucking reason (literaly only jumpsuits) okay wtf it fixed itself
- [x] Adds charging stations and find sprites for them
- [x] Add more cool interactions with electricity based objects
- [x] Add ethereal specific meat and blood
- [x] Map in ethereal specific objects (although I probably will make mappers do it because I suck at mapping)
- [x] Female sprites? (Or keep them agender? potential topic to discuss)
- [x] Sprites for alert for lowcharge (could potentially use other alerts)
- [x] Add sounds for burn attack

**also shill time:** https://www.patreon.com/user?u=14432003

### CODE REVIEW

**Comment by optimumtact** [comment]
I approved this PR to be made

**Comment by plapatin** [comment]
didn't we remove catgirls because we didnt want more snowflake races? why was this added

**Comment by HydroFloric** [comment]
@plapatin catpeople is getting axed because its a lazy race. Its literally just a human with a tail and ears. This is something actually interesting like a plasmameme.

**Comment by Qustinnus** [comment]
I think the main problem people have with catpeople is that theyre a reference to anime, are often over-sexualized in the media they're from (anime) and are essentialy just human + animal.

I don't think this is a snowflake race, but instead a cool sci-fi race.

**Comment by Anonymous583** [comment]
If this is added, friendship ended with humans. This sounds fucking fantastic.

**Comment by plapatin** [comment]
not saying i dont think this is a great idea. just weird timing with the catgirl thing

**Comment by Qustinnus** [comment]
i'm filling the hole with something i guess

**Comment by Anonymous583** [comment]
Would it be possible if they could simply have a normal light tint as an customization option instead of them being exclusively green-red color coded? e.g. they start off the same color as a light tube essentially and dim as their health/charge dips.

**Comment by Kierany9** [comment]
>naming your aliens ethereals

>not at all related these fuckers

ayy lmao

How disappointing

**Comment by Qustinnus** [comment]
@Kierany9 i named them ethereal because they shine like stars and seem otherworldy


i aint making them like those nerds that wipe my entire squad smh

**Comment by Qustinnus** [comment]
@Anonymous583 I'll look into more colors but I'd lock it to a few set ones to prevent people from pulling rainbow color shit on me

**Comment by QualityVan** [comment]
Should the nutrition changes be atomized into a separate PR? At a glance I'm not sure how entwined those changes are with eths.

**Comment by Qustinnus** [comment]
i made them specifically because i override nutrition adjustment for this species.

**Comment by Kierany9** [comment]
>They are walking lights; and shine bright and green if healthy, and get greyer and darker the more damaged they are. When dead they are just a grey unlit corpse.

>Sprites for alert for lowcharge (could potentially use other alerts)

How about they turn red when low on charge?

~~And once they hit critical levels, play the windows XP shutdown noise before falling asleep until being recharged~~

**Comment by Qustinnus** [comment]
I'm considering playing with the colors and different variables to try and get the nicest effect

**Comment by Qustinnus** [comment]
lol forgot my tag again

**Comment by Wilchenx** [comment]
possbility on zap effect when they hit someone with melee?

can it eat/drink plasma for fuel?

ability to charge themselves with power cells?

**Comment by Qustinnus** [comment]
@Wilchenx zap effect as in you see their skelebro for a second? probably not.

I like the second one; I'd say raw plasma would not work too well but it'd work; but processing it in food would be hella good

third; probably not.

**Comment by Qustinnus** [comment]
also, what do people think of agender race? should these guys have genders? thumb up/down to vote i guess and then I'll decide based on general opinion

**Comment by Anonymous583** [comment]
@Qustinnus assuming thumbs up means "yes" for genders, gotta say yes

**Comment by Qustinnus** [comment]
read the comment; up for agender, down for 2 genders

**Comment by Mickyan** [comment]
would it be too much if they could work as inducers by punching machines and losing their own charge

**Comment by Qustinnus** [comment]
probably too much yes.

**Comment by Jordie0608** [comment]
Get them fully working before worrying about the small details and fancy race specific functionality. Both can be smoothed over later.

**Comment by Qustinnus** [comment]
agreed, I'm going with only core gameplay for now (everything listed in OP) small interactions can be added later to prevent feature creep

**Comment by kriskog** [comment]
How about interaction with nightmares? Light eater usually snuffs all lights in first hit. Do they glow at all when dead? Worth thinking about.

**Comment by Qustinnus** [comment]
they dont glow at all when dead but i suppose they do counter nightmare a bit.

**Comment by ExcessiveUseOfCobblestone** [comment]
Depends on what happens when a lighteater hits them.

**Comment by Borisvanmemes** [comment]
I deeply like the idea of these but not how they look right now, shading seems off to me and the green is a tad dull imo.

**Comment by Qustinnus** [comment]
feel free to change up the sprites; I'm personally not a spriter and I havnt spoken to the guy who made these sprites since he made them.

**Comment by Ziiro** [comment]
How do they feel about radiation storms?

**Comment by Qustinnus** [comment]
currently they respond to it like any other species. I think I'm going to go with organic creatures that are fueled by electricity in their blood, this makes them shine and hot to the touch however this can be changed. I'm mainly doing this because non-organic seems like a pain to code right

File: code/modules/mob/living/carbon/human/species_types/ethereal.dm
Comment (Ziiro) [inline code/modules/mob/living/carbon/human/species_types/ethereal.dm]: don't null this, use "". Like Plasmemes:

https://github.com/tgstation/tgstation/blob/1b50671ca072dd646e98bee971a6d20d5c06a2de/code/modules/mob/living/carbon/human/species_types/plasmamen.dm#L20

**Comment by davethwave** [comment]
Perhaps instead of the power transfer from you to machine, you could get a implant/device that are made for Ethereals that would allow them to do that.

**Comment by davethwave** [comment]
Also, in my opinion they are a bit too bright, tone it down by one please.

File: code/modules/mob/living/carbon/human/species_types/ethereal.dm
Comment (Qustinnus) [reply code/modules/mob/living/carbon/human/species_types/ethereal.dm]: fixed thanks

File: code/modules/mob/living/carbon/human/species_types/ethereal.dm
Comment (Ziiro) [inline code/modules/mob/living/carbon/human/species_types/ethereal.dm]: What's the actual upper/lower ends of this at high/low charge?

**Comment by ike709** [comment]
recharging an ethereal with an inducer when

**Comment by davethwave** [comment]
*eyeroll

File: code/modules/mob/living/carbon/human/species_types/ethereal.dm
Comment (Qustinnus) [reply code/modules/mob/living/carbon/human/species_types/ethereal.dm]: 87.5% extra blunt if you're low charge
25% extra blunt if youre normal or higher charge.

**Comment by BeeSting12** [comment]
fetish content

**Comment by Qustinnus** [comment]
@BeeSting12 shitposting on a good faith pr will surely help catpeople

**Comment by wesoda25** [comment]
If you emag one it should go disco mode for a minute, possibly be confused.

**Comment by BeeSting12** [comment]
shocking is a fetish

**Comment by Wilchenx** [comment]
⚡️ 😍 ⚡️

**Comment by Borisvanmemes** [comment]
> If you emag one it should go disco mode for a minute, possibly be confused.

They aren't robots but this is too good of a idea to not get added tbh.

**Comment by BeeSting12** [comment]
what are the naming conventions for this race

**Comment by Qustinnus** [comment]
one ethereal
an ethereal spear
a group of atheari

i guess?

**Comment by Therapist1** [comment]
Maybe make their blood a weaker form of teslium? also rip space orcs

**Comment by BebeYoshi** [comment]
Do they have same naming as human (Caua Rebelo) or do they have special names like Plasmamen and Lizards? (Mercury XXI, Runs-In-Circles)

**Comment by Qustinnus** [comment]
i guess special names but i havnt come up with a convention so storm off to NTR hut and brainstorm for the perfect convention

**Comment by BeeSting12** [comment]
@Qustinnus thats what i meant, not what you'd call them

**Comment by Qustinnus** [comment]
I figured as much when i read bebeyoshi's comment

**Comment by wesoda25** [comment]
Lets name them after lightbulbs/wavelengths. Ulraviolet-27. 100 watt-78.

**Comment by MrDoomBringer** [comment]
hell yes @wesoda25 
also can we resprite them to be not green? I feel like having them look like ~space gas cowboys~ novakids could be cool
image

**Comment by Qustinnus** [comment]
@MrDoomBringer once i figure out how to make mcolor less gay and if the prefs menu works with me. I might make multicolor shit a seperate PR and limit it to some set colors to prevent obnoxious colors from annoying players

**Comment by Steelpoint** [comment]
So long as their naming convention is beyond the retarded Lizard people naming of "Breaks-the Immersion" or "Loots-the-Armoury" I'd be fine.

**Comment by Qustinnus** [comment]
thats a great idea @Steelpoint bad roman titles such as
Killius Catpersonus
Suqon Desius
Lootus Armorius
Dabius Onthemes

**Comment by Qustinnus** [comment]
also without memes heres a NTR thread for brainstorming kek https://tgstation13.org/phpBB/viewtopic.php?f=55&t=19975&view=unread#unread

**Comment by Grazyn** [comment]
What would a Nightmare do to them? Instakill? 
Also they shouldn't be able to take the nyctophobia trait because it would be a free trait for them

**Comment by kriskog** [comment]
>  also no one seem to complain that blood deficit is a free trait for plasmeme

Go make an issue report instead of using it to justify leaving other broken stuff in. Free bad traits are clearly not intended, and should be avoided in future code.

**Comment by Tlaltecuhtli** [comment]
nynphomania trait belongs to its own issue too, as the quirk system needs a species filer not here

File: code/__DEFINES/DNA.dm
Comment (XDTM) [inline code/__DEFINES/DNA.dm]: TRAIT_NOHUNGER already exists

**Comment by kingoftheautism** [comment]
When we are going to have our lanius bois?

File: code/__DEFINES/DNA.dm
Comment (Pantherfist) [reply code/__DEFINES/DNA.dm]: It has different behavior as far as I could tell

File: code/modules/mob/living/carbon/human/human_defense.dm
Comment (fludd12) [reply code/modules/mob/living/carbon/human/human_defense.dm]: Just make a new component and have the species assign that component on spawn - unless you mean something else.

**Comment by granpawalton** [comment]
Do you think using their charging as the cyborg or mech charging areas would be a good idea?

Also this is what new races should be like, plus id like to see someone try to pervert being lightning

File: code/modules/mob/living/carbon/human/human_defense.dm
Comment (XDTM) [reply code/modules/mob/living/carbon/human/human_defense.dm]: call RegisterSignal(mob, signal, proc) on the on_species_gain proc and unregister it on on_species_loss

File: code/__DEFINES/DNA.dm
Comment (XDTM) [reply code/__DEFINES/DNA.dm]: It probably constantly hardsets hunger to just below well fed; you can still keep your code and replace the species trait check with a has_trait(TRAIT_NOHUNGER) check, which would work for other sources of NOHUNGER as well.

**Comment by Vidiveni** [comment]
Teslium should make them shoot short range tesla arcs from their bodies, instead of just shocking them.

File: code/__DEFINES/DNA.dm
Comment (Ziiro) [reply code/__DEFINES/DNA.dm]: You'll probably want to make `UNUSUALHUNGER` or something, seeing as `NOHUNGER` implies that they just bypass the system altogether.

Comment (Qustinnus) [reply code/__DEFINES/DNA.dm]: It does bypass the system altogether

**Comment by Barhandar** [comment]
Lanius with full atmos integration when

**Comment by SouDescolado** [comment]
Holding light bulbs should turn the light bulbs on. Holding light bulbs that has plasma inside them make it explode. Bypassed if the user is wearing gloves

**Comment by SouDescolado** [comment]
Drinking Grey Bull (in case https://github.com/tgstation/tgstation/pull/40982 gets merged) should make them stop emitting light.

**Comment by ExcessiveUseOfCobblestone** [comment]
why?

**Comment by davethwave** [comment]
I guess considering this makes people resistant to electricity for a period of time, it would act like lipolicide.

**Comment by davethwave** [comment]
If the don't I would recommend that they make them more hungry as well as knocking for about 15s more or less depending on emp severity

**Comment by Tlaltecuhtli** [comment]
they arent machines

**Comment by SouDescolado** [comment]
> 
> 
> why?

Flavor, mostly. If something prevents energy go through your body, and this is a race made of energy basically, it should have a special interaction, I think

**Comment by Bawhoppen** [comment]
This is the *real* fetish content. Electrophiles will be coming in droves after this is merged.

**Comment by SouDescolado** [comment]
Everything is fetish content if you check it hard enough.

(I bet goof have a removal fetish)

**Comment by ExcessiveUseOfCobblestone** [comment]
can't spell /tg/ without the g used in goof

**Comment by Tlaltecuhtli** [comment]
jelly fish dont die to emps and i think emp should be only cause effects on robo shit, its already everywhere

**Comment by Qustinnus** [comment]
I'ma make the emag do some minor temporary memes but thats probably it

**Comment by granpawalton** [comment]
Jelly fish dont shock people, they have microscopic harpoon like structures with a venom, electric eels dont live off electricity either they generate it with an organ when needed. 

A minor emp interaction like nausea or a simple charge reduction would make sense and be a cool interaction even if the other downsides would have to be turned down imo

Also whats their interaction with disease/nausea?

**Comment by XDTM** [comment]
all you'd have to do is edit the species biotype, but organic makes sense all things considered

**Comment by Qustinnus** [comment]
I realize that but even without that flag their behavior would have to be changed a lot to fit not being classed as organic

**Comment by Qustinnus** [comment]
I'm going to work on the missing stuff tommorow. currently out of town so cant dev

**Comment by MrDoomBringer** [comment]
image
add shock juice reagent that shocks normal people and provides nutrients to these boys

**Comment by Qustinnus** [comment]
@MrDoomBringer bro dont just steal my DMs and put them in the pr

File: code/modules/mob/living/carbon/human/human_defense.dm
Comment (Qustinnus) [reply code/modules/mob/living/carbon/human/human_defense.dm]: I looked into this and I still think this is the better solution for now

**Comment by Qustinnus** [comment]
a heavily requested bonus because I was bored:
emp and emag interaction. Not really powerful but still a fun easter egg.
https://streamable.com/iw47w

**Comment by Qustinnus** [comment]
https://i.gyazo.com/677de02db7f30d0782ca2e62ec5c5092.mp4 names

**Comment by FantasticFwoosh** [comment]
I dont really have much in the way of criticism against ethereals besides them looking like goofy christmas lights with that emag interaction (i hope that can be fixed IC'ly so no ahelps to stop it) but the only thing grating my gears is that recharge pads are redundant when Ninja's have energy siphoning gloves that power abilities.

- Why not just let ethereals safely draw over wires/APC's/charged batteries passively with a ability

**Comment by Arianya** [comment]
Also shouldn't their bloodtype be LION, not LE? :^)

**Comment by Arianya** [comment]
> Why not just let ethereals safely draw over wires/APC's/charged batteries passively with a ability

This would either trivialize hunger for them (if it had very little effect on the actual charge of the APC/cell) or make them able to grief certain departments by sucking all the juice out of them

**Comment by Arianya** [comment]
> Why not just let ethereals safely draw over wires/APC's/charged batteries passively with a ability

This would either trivialize hunger for them (if it had very little effect on the actual charge of the APC/cell) or make them able to grief certain departments by sucking all the juice out of them

**Comment by Qustinnus** [comment]
@Jordie0608 chill out gamer

**Comment by FantasticFwoosh** [comment]
> 
> 
> > Why not just let ethereals safely draw over wires/APC's/charged batteries passively with a ability
> 
> This would either trivialize hunger for them (if it had very little effect on the actual charge of the APC/cell) or make them able to grief certain departments by sucking all the juice out of them

They can still get alternative foods from the chef in a later PR, there will be bajillion volts going through the system anyway leading to the charging pad, and griefing why not? Makes them unable to eat in the middle of a power sink and they might have a faster metabolism than other races losing charge.

**Comment by Qustinnus** [comment]
dm me the details but im kinda skeptical of too easy accesible methods that are too fast

**Comment by Qustinnus** [comment]
https://i.gyazo.com/756f346a1d615948025183c379a809a4.mp4

much requested feature ^

**Comment by Qustinnus** [comment]
i hate icon conflicts ree

**Comment by Wilchenx** [comment]
how do you do fellow greytiders

image

t. etheral

**Comment by Borisvanmemes** [comment]
I'd say make them grayscale but uhh, not sure that's doable.

**Comment by Qustinnus** [comment]
they are grayscale

**Comment by Borisvanmemes** [comment]
whole life is a lie

**Comment by Qustinnus** [comment]
i just dont want people picking from a color slider because a lot of colors can be annoying and would ruin the race due to a few bad apples picking neon pink

**Comment by fludd12** [comment]
I'd pick neon pink and pull it off just saying

File: code/modules/client/preferences.dm
Comment (Mickyan) [inline code/modules/client/preferences.dm]: could you possibly make this more generic if it's not too much trouble? Other races might be able to use something like this (flypeople, moths)

Comment (Qustinnus) [reply code/modules/client/preferences.dm]: mcolor

Comment (Mickyan) [reply code/modules/client/preferences.dm]: isn't mcolor picking a color directly from the full palette?

Comment (Mickyan) [reply code/modules/client/preferences.dm]: Maybe I'm reading it wrong but this is in
`if((DYNCOLORS in pref_species.species_traits))`
so if I set another race with DYNCOLORS I'd be getting the "Ethereal Colors" menu

What I meant was having the same thing except being able to pick from a different list depending on the race

Comment (Qustinnus) [reply code/modules/client/preferences.dm]: i can look into that potentialy

**Comment by plapatin** [comment]
i changed my mind. friendship ended with lizards, now ethereal is my best friend

**Comment by Anonymous583** [comment]
𝕗𝕒𝕥𝕙𝕖𝕣 𝕚 𝕕𝕖𝕤𝕚𝕣𝕖 𝕥𝕙𝕖 𝕄𝔼ℝ𝔾𝔼

**Comment by MACIEKBAKI** [comment]
Will the permabrig area have the food for these guys? I mean what if they get thrown into perma ~~which is impossible we all know perma is useless~~ do they just starve there?

**Comment by Qustinnus** [comment]
no they can touch the lights

**Comment by Qustinnus** [comment]
hello maintainers i am a cute boy from the netherlands looking for a single maintainer to give me a review <3

**Comment by optimumtact** [comment]
you're not going to get anything until the poll ends so you may as well continue looking cute but more silently

**Comment by Qustinnus** [comment]
image

**Comment by bobbahbrown** [comment]
hey its me the maintainers i approve this pr but i have misplaced my merge button ill have to delegate that

**Comment by Qustinnus** [comment]
thanks, the maintainers.

File: code/game/machinery/rechargestation.dm
Comment (AnturK) [inline code/game/machinery/rechargestation.dm]: What's the point of null conditional if you're going to runtime on next line anyway.

File: code/modules/mob/living/carbon/human/human_defense.dm
Comment (AnturK) [reply code/modules/mob/living/carbon/human/human_defense.dm]: Should be species proc at least.

**Review by AnturK — CHANGES_REQUESTED** [review]
Besides these, No hunger should be handled by the mob not the caller in my opinion.

File: code/game/machinery/rechargestation.dm
Comment (Qustinnus) [reply code/game/machinery/rechargestation.dm]: because im retarded

**Comment by optimumtact** [comment]
@Qustinnus address reviews but this is cleared for merge

**Comment by Qustinnus** [comment]
I'm still in Denmark I'll get home this week and then I can address the final stuff

File: code/modules/client/preferences.dm
Comment (Qustinnus) [reply code/modules/client/preferences.dm]: If I get the features after setting the species, init_species isn't being ran properly. (It'll use the old features)

Comment (Qustinnus) [reply code/modules/client/preferences.dm]: whoever moved it above that is a chumbis

**Comment by Qustinnus** [comment]
pr done also subtle shill https://www.patreon.com/user?u=14432003 for if this gets merged and you actually dont hate my guts :^)

**Comment by optimumtact** [comment]
`high quality prs`

**Comment by Qustinnus** [comment]
relative terms

**Comment by moonheart08** [comment]
Resprite to colored lightbulbs for christmas.

**Comment by AlexTheSysop42** [comment]
Their heirloom should be a flashsuit like a lamp is for moths

**Comment by Qustinnus** [comment]
testmerge showed some bugs so im working on fixing those.

**Comment by Zaffinomlb** [comment]
We don't need any more races. Honestly you are removing the charm of the existing ones by over saturating the existing pool.
This is a mistake.

**Comment by davethwave** [comment]
A BIT late on that, we even removed a race to allow this one in...

**Comment by TheRespectedMan** [comment]
@Zaffinomlb That's why we're removing the chinese cartoon catgirl race from the config to make room for a better race. There's room now.

**Comment by MrDoomBringer** [comment]
@davethwave [its a meme silly](https://knowyourmeme.com/memes/chinese-cartoons)

**Comment by davethwave** [comment]
Huh, never knew there was such a meme.

**Comment by Shadowflame909** [comment]
First, there were moffs. And now there are living light sticks. Will the shadow people and the nightmares ever be able to survive the wrath of races attracted to light?

ow.

**Comment by Borisvanmemes** [comment]
How much longer must we wait for glowy bois to get merged?

**Comment by adamkad1** [comment]
make them explode if they die, or give them ability to charge apcs, aaaa

**Comment by Qustinnus** [comment]
@adamkad1 feature creep, itll b e in a future pr

**Comment by Shadowflame909** [comment]
Tesla, Tasers, Tesla Guns should just make their nutrition fill to the max and make them fat.

But being fat should have intense downsides to them. Like double the slowdown a normal fat human gets.

Just some ideas for mechanical differences on how stuns affect humans compared to Ethereals.

**Comment by Trazen4** [comment]
For some reason changing moth wings doesn't work, and all moths spawn with the default wing style.

**Comment by Qustinnus** [comment]
This is something with prefs and I don't know what. I think the save files are fucking themselves

**Comment by AlexTheSysop42** [comment]
Being fat should make them run slightly faster than normal but also constantly take damage, since they're basically being overloaded

**Comment by Mickyan** [comment]
I have reason to believe this breaks wing selection for moths, it no longer works on bagil where this is testmerged, but it still works on local and sybil/terry

**Comment by Qustinnus** [comment]
ughh I fucking hate prefs. it works fine locally so I'll need someone who knows about prefs @kevinz000 @XDTM send help

**Comment by ExcessiveUseOfCobblestone** [comment]
i'm not a maintainer but can I still cash in on the offer?

File: code/modules/mob/living/simple_animal/slime/life.dm
Comment (Qustinnus) [reply code/modules/mob/living/simple_animal/slime/life.dm]: Already fixed on local

**Pull request merged**
